detekt-gradle-plugin/src/main/kotlin/dev/detekt/gradle/plugin/internal/DetektKmpJvmCompilations.kt

```
@@ -0,0 +1,21 @@
+package dev.detekt.gradle.plugin.internal
+
+import io.gitlab.arturbosch.detekt.extensions.DetektExtension
+import org.gradle.api.Project
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType.androidJvm
+import org.jetbrains.kotlin.gradle.plugin.KotlinPlatformType.jvm
+import org.jetbrains.kotlin.gradle.plugin.KotlinTargetsContainer
+
+internal object DetektKmpJvmCompilations {
+    fun registerTasks(project: Project, extension: DetektExtension) {
+        val kotlinExtension = project.extensions.getByName("kotlin")
+        if (kotlinExtension !is KotlinTargetsContainer) return
```

Same here. Is this expected? Should we just throw an exception instead so if this happen we get faster and easier reports to fix it?

---

Yeah at this point we already know multiplatform plugin is applied and the multiplatform extension implements `KotlinTargetsContainer` so that will be safe to do. I was being defensive here, if there's an issue with a new Kotlin version our tests will expose it, but there's no harm making it an exception.